javascript/src/events/event-reporter.ts

```diff
@@ -8,48 +10,39 @@ import { Logger } from "../utils/logger";
  * with proper authentication and error handling.
  */
 export class EventReporter {
-  private readonly eventsEndpoint: URL;
   private readonly apiKey: string;
+  private readonly eventsEndpoint: URL;
+  private readonly eventAlertMessageLogger: EventAlertMessageLogger;
   private readonly logger = new Logger("scenario.events.EventReporter");
 
   constructor(config: { endpoint: string; apiKey: string | undefined }) {
-    this.eventsEndpoint = new URL("/api/scenario-events", config.endpoint);
     this.apiKey = config.apiKey ?? "";
-
-    if (!process.env.SCENARIO_DISABLE_SIMULATION_REPORT_INFO) {
-      if (!this.apiKey) {
-        console.log(
-          "➡️  LangWatch API key not configured, simulations will only output the final result"
-        );
-        console.log(
-          "To visualize the conversations in real time, configure your LangWatch API key (via LANGWATCH_API_KEY, or scenario.config.js)"
-        );
-      } else {
-        console.log(`simulation reporting is enabled, endpoint:(${this.eventsEndpoint}) api_key_configured:(${this.apiKey.length > 0 ? "true" : "false"})`);
-      }
-    }
+    this.eventsEndpoint = new URL("/api/scenario-events", config.endpoint);
+    this.eventAlertMessageLogger = new EventAlertMessageLogger();
+    this.eventAlertMessageLogger.handleGreeting();
   }
 
   /**
    * Posts an event to the configured endpoint.
    * Logs success/failure but doesn't throw - event posting shouldn't break scenario execution.
    */
-  async postEvent(event: ScenarioEvent): Promise<void> {
-    this.logger.debug(`[${event.type}] Posting event`, {
-      event,
-    });
+  async postEvent(event: ScenarioEvent): Promise<{ setUrl?: string }> {
+    const result: { setUrl?: string } = {};
+    this.logger.debug(`[${event.type}] Posting event`, { event });
 
-    if (!this.eventsEndpoint) {
+    if (!env.LANGWATCH_ENDPOINT) {
```

Comment:
i think this isn't possible as it has a default

Reply:
good catch.